src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -370,8 +370,8 @@ public void OnResponseHeadersComplete(bool endStream)
                     {
                         if (!endStream)
                         {
-                             if (NetEventSource.IsEnabled) Trace("Trailing headers received without endStream");
-                             throw new Http2ProtocolException(SR.net_http_invalid_response);
+                            if (NetEventSource.IsEnabled) Trace("Trailing headers received without endStream");
+                            throw new Http2ConnectionException(Http2ProtocolErrorCode.ProtocolError);
```

Comment:
How about including a message along with the ProtocolError that indicates what specifically was the problem? For here and other such cases.

Reply:
I agree this would be a good thing to do; I don't think it needs to be part of this PR. We have a lot of places where we throw Http2ConnectionException today, and none of them have a custom message. We should probably go through all of these and try to add more detailed messages.

That said, all of these cases (should) mean the server did something really bad, and so hopefully are rare.